src/SessionSetup/ConnectToLocalWidget.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "SessionSetup/ConnectToLocalWidget.h"
+
+#include <absl/flags/flag.h>
+#include <absl/time/time.h>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/security/credentials.h>
+#include <grpcpp/support/channel_arguments.h>
+
+#include <QMessageBox>
+#include <QPushButton>
+#include <algorithm>
+#include <memory>
+
+#include "ClientFlags/ClientFlags.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Result.h"
+#include "SessionSetup/Connections.h"
+#include "ui_ConnectToLocalWidget.h"
+
+namespace orbit_session_setup {
+
+// The destructor needs to be defined here because it needs to see the type
+// `Ui::ConnectToLocalWidget`. The header file only contains a forward declaration.
+ConnectToLocalWidget::~ConnectToLocalWidget() = default;
+
+ConnectToLocalWidget::ConnectToLocalWidget(QWidget* parent)
+    : QWidget(parent),
+      ui_(std::make_unique<Ui::ConnectToLocalWidget>()),
+      local_connection_(
+          grpc::CreateCustomChannel(absl::StrFormat("127.0.0.1:%d", absl::GetFlag(FLAGS_grpc_port)),
+                                    grpc::InsecureChannelCredentials(), grpc::ChannelArguments()),
+          nullptr),
+      check_connection_timer_(this) {
+  ui_->setupUi(this);
+
+  // The following is to make the radioButton behave as if it was part of an exclusive button group
+  // in the parent widget (SessionSetupDialog). If a user clicks on the radioButton and it was
+  // not checked before, it is checked afterwards and this widget sends the activation signal.
+  // SessionSetupDialog reacts to the signal and deactivates the other widgets belonging to that
+  // button group. If a user clicks on a radio button that is already checked, nothing happens, the
+  // button does not get unchecked.
+  QObject::connect(ui_->radioButton, &QRadioButton::clicked, this, [this](bool checked) {
```

Comment:
That seem to be complicated, but I guess its okay.

Reply:
You are right, this is too complicated. This specific part is copied from LoadCaptureWidget and at this point pure code duplication. I will change this soon, to be at least deduplicated, but more likely simply go away.